ansible/roles/airflow/files/airflow-dags/canning.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+import subprocess
+from airflow import DAG
+from airflow.exceptions import AirflowException # signal ERROR
+from airflow.operators.bash_operator import BashOperator
+from airflow.operators.sensors import BaseSensorOperator
+from datetime import datetime, timedelta
+
+class ReportsRawReadySensor(BaseSensorOperator):
+    def poke(self, context):
+        retcode = subprocess.call(['sudo', '--non-interactive', '/usr/local/bin/docker-trampoline', 'is-reports-raw-ready',
+            context['ds'], context['execution_date'].isoformat(), (context['execution_date'] + context['dag'].schedule_interval).isoformat()])
+        if retcode == 42:
+            return True
+        elif retcode == 13:
+            return False
+        else:
+            raise AirflowException('Unexpected `is-reports-raw-ready` exit code: {:d}'.format(retcode))
+
+dag = DAG(
+    dag_id='hist_canning',
+    schedule_interval=timedelta(days=1),
+    start_date=datetime(2012, 12, 5),
+    end_date=datetime(2017, 4, 22), # NB: end_date is included
```

Comment:
It's the last processed date in the schedule. I've not enabled recurring scheduling of the tasks as I was going offline for a while. It'll be commented when pipeline starts ticking daily.